ocaml/xapi-cli-server/record_util.ml

```diff
@@ -628,6 +628,49 @@ let string_to_on_crash_behaviour s =
            )
         )
 
+let on_softreboot_behaviour_to_string x =
```

Comment:
you could share code here with on_crash to string. Handle the specific new case first ('soft_reboot), and then on _ hand it over to the other.

Reply:
I tried to do this, but it doesn't seem to be liking it, it is using an anonymous type and doesn't like to mix `\`soft_reboot` with the others.

Reply:
Does something like this work?
```
match x with
| `soft_reboot -> "Soft reset"
| rest -> on_crash_behaviour_to_string rest
```

Reply:
Does it work?

Reply:
I meant: does Edwin's suggestion to simplify the code work?

Reply:
I did try to do this, however, due to the one extra variant, OCaml doesn't think the types are the same.